code/modules/clothing/head/winter_hats.dm

```diff
@@ -50,15 +50,25 @@
 	. = ..()
 	color = get_random_colour(lower = 150)
 
-/obj/item/clothing/head/beanie/winter
-	name = "winter beanie"
+/obj/item/clothing/head/beanie/earflap
+	name = "ear flap beanie"
 	desc = "A head-hugging brimless winter cap. This one has flaps that cover the ears."
-	icon_state = "winterbeanie"
-	item_state = "winterbeanie"
+	icon_state = "beanie_earflap"
+	item_state = "beanie_earflap"
 	has_accents = TRUE
 
+/obj/item/clothing/head/beanie/earflap/random/Initialize()
+	. = ..()
+	color = get_random_colour(lower = 150)
+	accent_color = color
+
 /obj/item/clothing/head/beanie/submariner
 	name = "submariner's beanie"
 	desc = "A design of tightly fitting beanie particularly popular among the dock workers of Europa. Favored among anyone who prides a warm head."
-	icon_state = "dock_beanie"
-	item_state = "dock_beanie"
+	icon_state = "beaner_submariner"
+	item_state = "beaner_submariner"
+
+/obj/item/clothing/head/beanie/submariner/random/Initialize()
+	. = ..()
+	color = get_random_colour(lower = 150)
```

Comment:
you should remove this or add a colorable and random subtype at minimum, i wouldn't want to use a loadout item that spawns with  a random colour

Reply:
this isn't a loadout item it's an outfit item for actors, the default is the colorable version